WordPress/src/main/java/org/wordpress/android/util/ContentDescriptionListAnnouncer.kt

```diff
@@ -0,0 +1,67 @@
+package org.wordpress.android.util
+
+import android.view.View
+import android.view.accessibility.AccessibilityEvent
+import android.view.accessibility.AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED
+import android.view.accessibility.AccessibilityNodeInfo.ACTION_CLICK
+import androidx.annotation.StringRes
+import androidx.core.view.AccessibilityDelegateCompat
+import androidx.core.view.accessibility.AccessibilityNodeInfoCompat
+
+class ContentDescriptionListAnnouncer constructor(
+    @StringRes private val emptyListText: Int,
+    @StringRes private val endOfListText: Int,
+    @StringRes private val clickActionText: Int? = null,
+    private val contentDescriptions: List<String>,
+    targetView: View
+) {
+    private var currentIndex = 0
+
+    init {
+        AccessibilityUtils.setAccessibilityDelegateSafely(
+                targetView,
+                object : AccessibilityDelegateCompat() {
+                    override fun onPopulateAccessibilityEvent(
+                        host: View?,
+                        event: AccessibilityEvent?
+                    ) {
+                        super.onPopulateAccessibilityEvent(host, event)
+                        if (event?.eventType == TYPE_VIEW_ACCESSIBILITY_FOCUSED) {
+                            currentIndex = 0
+                        }
+                    }
+
+                    override fun onInitializeAccessibilityNodeInfo(
+                        host: View?,
+                        info: AccessibilityNodeInfoCompat?
+                    ) {
+                        super.onInitializeAccessibilityNodeInfo(host, info)
+
+                        clickActionText?.let {
+                            info?.addAction(
+                                    AccessibilityNodeInfoCompat.AccessibilityActionCompat(
+                                            ACTION_CLICK, host?.context?.getString(
```

Comment:
In this class, I utilize the `context` of the host view so I don't have to inject the `ResourceProvider` since this class isn't testable and doesn't need to be right now since it's just a view related one. What do you think?

Reply:
Yep, I agree that it's ok => the class isn't testable anyway and making it testable would require more changes.